crates/pallets/starknet/runtime_api/src/lib.rs

```diff
@@ -1,5 +1,6 @@
 //! Definition of the runtime API for the StarkNet pallet.
 
+#![cfg_attr(not(feature = "std"), no_std)]
```

Comment:
Here is the thing that was missing. It make sure that the std lib is not imported if the feature `std` is not present